### PR TITLE
make dbname / database optional to provide in the input profiles

### PR DESCRIPTION
**Pull request by tovganesh**
make dbname / database optional to provide in the input.
if provided, ensure dbname / database has the same value as schema.
this is because in impala schema and database are the same

Internal Ticket: https://jira.cloudera.com/browse/DBT-120

Testplan:
a) Use a profile something like below with dbt debug (only provide schema)
<pre>
dbtdemo:
  target: dev_impala_ldap
  outputs:
   dev_impala_ldap:
     type: impala
     host: dbt-impala-gateway.ciadev.cna2-sx9y.cloudera.site
     http_path: dbt-impala/cdp-proxy-api/impala
     username: [username]
     password: [password]
     port: 443
     auth_type: ldap
     auth: LDAP
     use_http_transport: true
     use_ssl: true
     schema: dbtdemo
     threads: 1
</pre>
b) Use a profile something like below with dbt debug (provide schema and dbname with same value)
<pre>
dbtdemo:
  target: dev_impala_ldap
  outputs:
   dev_impala_ldap:
     type: impala
     host: dbt-impala-gateway.ciadev.cna2-sx9y.cloudera.site
     http_path: dbt-impala/cdp-proxy-api/impala
     username: [username]
     password: [password]
     port: 443
     auth_type: ldap
     auth: LDAP
     use_http_transport: true
     use_ssl: true
     schema: dbtdemo
     dbname: dbtdemo
     threads: 1
</pre>
c) Use a profile something like below with dbt debug (provide schema and dbname with different values)
<pre>
dbtdemo:
  target: dev_impala_ldap
  outputs:
   dev_impala_ldap:
     type: impala
     host: dbt-impala-gateway.ciadev.cna2-sx9y.cloudera.site
     http_path: dbt-impala/cdp-proxy-api/impala
     username: [username]
     password: [password]
     port: 443
     auth_type: ldap
     auth: LDAP
     use_http_transport: true
     use_ssl: true
     schema: dbtdemo
     dbname: dbtdemo1
     threads: 1
</pre>

When run (a) and (b) should succeed and (c) should fail with the following error:
<pre>
Runtime Error
  Credentials in profile "dbtdemo", target "dev_impala_ldap" invalid: Runtime Error
        schema: dbtdemo
        database: dbtdemo1
    On Impala, database must be omitted or have the same value as schema.
</pre>

Finally, for (a) and (b) all the dbt commands (dbt run, dbt seed, dbt test, dbt docs generate and dbt docs serve) should run.

### CODE REVIEW

File: dbt/adapters/impala/connections.py [hunk 1]
```diff
@@ -60,14 +59,35 @@ class ImpalaCredentials(Credentials):
...
+    @classmethod
+    def __pre_deserialize__(cls, data):
+        data = super().__pre_deserialize__(data)
+        if 'database' not in data:
+            data['database'] = None
+        return data
+
+    def __post_init__(self):
+        # impala classifies database and schema as the same thing
+        if (
```
Comment (TapasSenapati) [inline dbt/adapters/impala/connections.py:71]: nitpick formatting. Add a linter to format this.

File: dbt/adapters/impala/connections.py [hunk 2]
```diff
@@ -60,14 +59,35 @@ class ImpalaCredentials(Credentials):
...
+            self.database is not None and
+            self.database != self.schema
+        ):
+            raise dbt.exceptions.RuntimeException(
+                f'    schema: {self.schema} \n'
+                f'    database: {self.database} \n'
+                f'On Impala, database must be omitted or have the same value as'
+                f' schema.'
+            )
```
Comment (TapasSenapati) [inline dbt/adapters/impala/connections.py:80]: Do we need a conditional check to throw exception if profile file is missing schema.

Comment (tovganesh) [reply dbt/adapters/impala/connections.py:80]: No, schema is a required parameter as opposed to database being optional and is already handled in the standard checks in the dbt flow.

File: dbt/adapters/impala/connections.py (hunk 1 above)
Comment (tovganesh) [reply dbt/adapters/impala/connections.py:71]: Not sure what you mean here, looks like this is the formatting I get from my linter :)

**Review by TapasSenapati — APPROVED** [review]
LGTM

**Pull request merged**